src/oop/programs_with_multiple_classes/medium/q8/MainTest.java

```diff
@@ -0,0 +1,137 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    private final String classLight = "Light";
+    private final String classKitchen = "Kitchen";
+    private final String varOn = "isOn";
+    private final String varType = "type";
+    private final String varSize = "size";
+    private final String varCap = "personCapacity";
+    private final String varLights = "lights";
+    public ObjectTest light;
+    public ObjectTest kitchen;
+
+    public Clause[] testSentence() {
+        return new Clause[0];
+    }
+
+    public void runMain() {
+    }
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String lightClassString = "oop.programs_with_multiple_classes.medium.q8." + classLight;
+        String kitchenClassString = "oop.programs_with_multiple_classes.medium.q8." + classKitchen;
+        light = new ObjectTest(lightClassString);
+        kitchen = new ObjectTest(kitchenClassString);
+        String modifiedLightMessage =
+                "You have modified the provided portions of class " + classLight + ". Please revert them to the original state.";
+        assertTrue(light.hasField(varOn, boolean.class), modifiedLightMessage);
+        assertTrue(light.hasModifier(varOn, "private"), modifiedLightMessage);
+        assertTrue(light.hasField(varType, String.class), modifiedLightMessage);
+        assertTrue(light.hasModifier(varType, "private"), modifiedLightMessage);
+        String modifiedKitchenMessage =
+                "You have modified the provided portions of class " + classKitchen + ". Please revert them to the original state.";
+        Class<?>[] classArguments = {int.class, int.class};
+        assertTrue(kitchen.hasConstructor(classArguments), modifiedKitchenMessage);
+        assertTrue(kitchen.hasModifier(classArguments, "public"), modifiedKitchenMessage);
+
+        Object[][] test = new Object[][]{{10, 15}, {20, 30}, {150, 40}};
+        for (int num = 0; num < test.length; num++) {
+            int size = (int) test[num][0];
+            int personCapacity = (int) test[num][1];
+            Object[][] arguments = {
+                    {size, int.class},
+                    {personCapacity, int.class}
+            };
+            Object kitchenInstance = kitchen.createInstance(arguments);
+            _assertEquals(size, kitchen.getFieldValue(kitchenInstance, varSize),
+                    "Your " + classKitchen + " constructor does not correctly initialize the " + varSize + " field.");
+            _assertEquals(personCapacity, kitchen.getFieldValue(kitchenInstance, varCap),
+                    "Your " + classKitchen + " constructor does not correctly initialize the " + varCap + " field.");
+        }
+    }
+
+    @Test
+    public void kitchenClassHasCorrectAttributes() {
+        String incorrectFieldMessage = "Your " + classKitchen + " class is missing a required field.";
```

Comment:
We can also switch to say which field is missing. I thought that at first, this would be too obvious, but they are beginners